Frontend/ui-driver/src/Screens/RegistrationScreen/View.purs

```diff
@@ -361,6 +374,46 @@ messageView push state =
           ]
       ]
 
+enterReferralCode :: forall w . (Action -> Effect Unit) -> ST.RegistrationScreenState -> PrestoDOM (Effect Unit) w
+enterReferralCode push state =
+  linearLayout
+          [ width MATCH_PARENT
+          , height WRAP_CONTENT
+          , orientation HORIZONTAL
+          , stroke $ "1," <> Color.grey900
+          , cornerRadius 4.0
+          , padding $ Padding 10 10 10 10
+          ][  textView $
+              [ width MATCH_PARENT
+              , height WRAP_CONTENT
+              , color if all (_ == COMPLETED) [ state.data.vehicleDetailsStatus, state.data.drivingLicenseStatus, state.data.permissionsStatus ] then Color.black900 else Color.greyTextColor
```

Comment:
`all (_ == COMPLETED) [ state.data.vehicleDetailsStatus, state.data.drivingLicenseStatus, state.data.permissionsStatus ]` 
This is being repeated. Use let in to store the output of this and then use it